src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/ISleepIfBusy.sql

```diff
@@ -16,96 +16,77 @@ BEGIN
+    IF (@@TRANCOUNT > 0)
+        THROW 50400, 'Cannot sleep in transaction', 1
+
+    WHILE (1 = 1)
     BEGIN
-        IF (@@TRANCOUNT > 0)
-            THROW 50400, 'Cannot sleep in transaction', 1
+        SELECT  @throttleCount = ISNULL(SUM(CASE
+                                                WHEN r.wait_type IN ('IO_QUEUE_LIMIT', 'LOG_RATE_GOVERNOR', 'SE_REPL_CATCHUP_THROTTLE', 'SE_REPL_SLOW_SECONDARY_THROTTLE', 'HADR_SYNC_COMMIT') THEN 1
+                                                ELSE 0
+                                            END), 0),
+                @sleepersCount = ISNULL(SUM(CASE
+                                WHEN r.wait_type IN ('WAITFOR') THEN 1
+                                ELSE 0
+                            END), 0),
+                @activeRequestCount = COUNT(*)
+        FROM    sys.dm_exec_requests r WITH (NOLOCK)
+        JOIN    sys.dm_exec_sessions s WITH (NOLOCK)
+        ON      s.session_id = r.session_id
+        WHERE   r.session_id <> @@spid
+                AND s.is_user_process = 1 -- user sessions only
 
-        DECLARE @computedThrottleActiveRequestCount INT
+        SET @activeRequestCount = @activeRequestCount - @sleepersCount
 
-        WHILE (1 = 1)
+        IF (@throttleCount > 0)
         BEGIN
-            SELECT  @throttleCount = ISNULL(SUM(CASE
-                                                    WHEN r.wait_type IN ('IO_QUEUE_LIMIT', 'LOG_RATE_GOVERNOR', 'SE_REPL_CATCHUP_THROTTLE', 'SE_REPL_SLOW_SECONDARY_THROTTLE', 'HADR_SYNC_COMMIT') THEN 1
-                                                    ELSE 0
-                                                END), 0),
-                    @sleepersCount = ISNULL(SUM(CASE
-                                    WHEN r.wait_type IN ('WAITFOR') THEN 1
-                                    ELSE 0
-                                END), 0),
-                    @activeRequestCount = COUNT(*)
-            FROM    sys.dm_exec_requests r WITH (NOLOCK)
-            JOIN    sys.dm_exec_sessions s WITH (NOLOCK)
-            ON      s.session_id = r.session_id
-            WHERE   r.session_id <> @@spid
-                    AND s.is_user_process = 1 -- user sessions only
-
-            SET @activeRequestCount = @activeRequestCount - @sleepersCount
-
-            IF (@throttleCount > 0)
-            BEGIN
-                RAISERROR('Throttling due to write waits', 10, 0) WITH NOWAIT
-                WAITFOR DELAY '00:00:02'
-            END
-            ELSE IF (@activeRequestCount >= 0)
-            BEGIN
-                SET @throttleActiveRequestCount = NULL
-
-                -- Check if we have a setting in the tbl_ResourceManagementSetting table.
-                -- We need to check this setting on every iteration of the loop, to make ensure that we can increase the value if prc_iSleepIfBusy is stuck.
-                SELECT  @throttleActiveRequestCount = CONVERT(INT, Value)
-                FROM    tbl_ResourceManagementSetting
```

Comment:
Embarrassingly I did not scrutinize this sproc more. It relies on a user table called `tbl_ResourceManagementSetting` to optionally (and dynamically) adjust the throttle. I think for our sake for now we can skip this part and simply compute it instead